packages/api-specification/api/windows.ts

```diff
@@ -0,0 +1,313 @@
+     */
+    show?: boolean;
+
+    /**
+     * Whether to show the window in taskbar. Default is false.
+     */
+    skipTaskbar?: boolean;
+
+    /**
+     * Makes the window transparent. Default is false.
+     */
+    transparent?: boolean;
+}
+
+/**
+ * A window running in the container.
+ * It implements EventEmitter which means users can subscribe for events using on, once , addEventListener etc.
+ * The list of events is in WindowsEvents structure;
+ */
+export interface Window extends EventEmitter {
+
+    /** Get or set the max width of the window. Might be undefined if no restrictions applied**/
+    maxWidth?: number;
+
+    /** Get or set the min width of the window. Might be undefined if no restrictions applied**/
+    minWidth?: number;
+
+    /** Get or set the max height of the window. Might be undefined if no restrictions applied**/
+    maxHeight?: number;
+
+    /** Get or set the min height of the window. Might be undefined if no restrictions applied**/
+    minHeight?: number;
+
+    /**
+     * Get or set if the window has a shadow.
+     */
+    hasShadow: boolean;
+
+    /**
+     * Get or set the title of the window
+     */
+    title: string;
+
+    /**
+     * Get or set if the window is always on top of all other windows.
+     */
+    alwaysOnTop: boolean;
+
+    /**
+     * Get or set if the window can be maximized.
+     */
+    maximizable: boolean;
+
+    /**
+     * Get or set if the window can be minimized.
+     */
+    minimizable: boolean;
+
+    /**
+     * Get or set if the window can be re-sized.
+     */
+    resizable: boolean;
+
+    /**
+     * Get or set if the window is shown in the taskbar.
+     */
+    skipTaskbar: boolean;
+
+    /**
+     * Removes focus from the window.
+     * @returns {Promise<void>} A promise which resolves to nothing when the function has completed.
+     */
+    blur(): Promise<void>;
+
+    /**
+     * Closes the window.
+     * @returns {Promise<void>} A promise which resolves to nothing when the function has completed.
+     */
+    close(): Promise<void>;
+
+    /**
+     * Flashes the window's frame and taskbar icon.
+     * @param {boolean} flag - Flag to start or stop the window flashing.
+     * @returns {Promise<void>} A promise which resolves to nothing when the function has completed.
+     */
+    flashFrame(flag: boolean): Promise<void>;
+
+    /**
+     * Activates the window. This will bring the window to front or restore it if minimized.
+     * If focus flag is true (default) the window will also receive input focus.
+     * @returns {Promise<void>} A promise which resolves to nothing when the function has completed.
+     */
+    activate(focus?: boolean): Promise<void>;
+
+    /**
+     * Captures an image of the window
+     * @param options
+     */
+    capture(options?: CaptureOptions): Promise<Base64ImageData>;
+
+    /** Get the bounds of the window */
+    getBounds(): Bounds;

```

Comment:
They are, although I do wonder whether we should wrap them to be synchronous? All the equivalent Electron methods are synchronous.

I wonder whether this is an OpenFin implementation detail that has 'leaked' into the public API?